src/ui/components/SearchInput/style.scss

```diff
@@ -0,0 +1,90 @@
+@import "~core/css/inc/mixins";
+
+$icon-margin: 16px;
+$icon-size: 26px;
+
+.SearchInput {
+  position: relative;
+  text-align: center;
+  width: 100%;
+
+  .Icon-magnifying-glass {
+    @include start(0);
+    height: 3rem;
+    margin: 0 $icon-margin;
+    opacity: 0;
+    position: absolute;
+    transition: opacity 1ms 250ms, transform 250ms;
+  }
+}
+
+.SearchInput--text {
+  @include text-align-start();
+
+  .SearchInput-label {
+    color: transparent;
+
+    &::before {
+      opacity: 0;
+      transition: none;
+    }
+  }
+
+  .Icon-magnifying-glass {
+    color: #fff;
+    opacity: 1;
+    transition: transform 250ms;
+
+    [dir=ltr] & {
+      transform: translateX(0) !important;
+    }
+
+    [dir=rtl] & {
+      // transform: translateX(-32px);
+    }
+  }
+}
+
+.SearchInput-label {
+  @include font-regular();
+  color: #ccc;
+  display: inline-block;
+  height: 3rem;
+  line-height: 3rem;
```

Comment:
I was setting this to centre the text vertically since the input bar is `3rem`s high. It might not be needed though, I'll at least extract this to a variable if it's still needed.